function/src/test/java/com/example/driver/DemoApplicationTests.java

```diff
@@ -1,27 +1,35 @@
 package com.example.driver;
 
-import static org.assertj.core.api.Assertions.assertThat;
-
 import org.junit.jupiter.api.Test;
+
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.test.web.reactive.server.WebTestClient;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+
+import static org.assertj.core.api.Assertions.assertThat;
 
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 public class DemoApplicationTests {
 
-	@Autowired
-	private WebTestClient webTestClient;
-
 	@Test
-	void match() {
-		webTestClient
-				.post().uri("/")
-				.header("one", "two")
-				.bodyValue("Hello World")
-				.exchange()
-				.expectStatus().isOk()
-				.expectHeader().value("one", value -> assertThat(value).isEqualTo("two"));
+	void match(@Autowired TestRestTemplate restTemplate) {
+		HttpHeaders headers = new HttpHeaders();
```

Comment:
Oh boy, going back from WebTestClient to TestRestTemplate is definitely a downgrade. I know that we can use WebTestClient w/ web mvc but I was running into missing beans even though I was pulling in `spring-webflux`. I chose to stop battling this and will switch back once we [switch back to webflux](https://github.com/dsyer/spring-wasm-demo/pull/3/files#r963217070).